### PR TITLE
test: make use of async/await

### PR DESCRIPTION
**Pull request by hiroppy**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [x] This is a **code refactor**
- [x] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?

modify only

### Motivation / Use-Case

We don't support Node6, so we can use async/await.

### Breaking Changes

no

### Additional Info

Next step, I'll refactor test/run-server and delete TODO.

### CODE REVIEW

**Comment by hiroppy** [comment]
I deleted test/Server.test.js because I failed to rebase from master to next.

File: .babelrc [hunk 1]
```diff
@@ -1,3 +1,8 @@
 {
-  "presets": ["@babel/preset-env"]
+  "presets": ["@babel/preset-env"],
+  "env": {
+    "test": {
+      "plugins": ["@babel/plugin-transform-runtime"]
```
Comment (hiroppy) [inline .babelrc]: I'm not sure but jest uses babel as the default, so we need `regeneratorruntime` to use async/await. If we want to use the native async/await, we have to set `current` using `@babel/env`.

File: client-src/clients/BaseClient.js [hunk 2]
```diff
@@ -1,5 +1,3 @@
-'use strict';
```
Comment (hiroppy) [inline client-src/clients/BaseClient.js]: Affected by `"sourceType": "module"` in `./client-src/.eslintrc`

**Comment by hiroppy** [comment]
Merge #1997 first.

File: .babelrc (hunk 1 above)
Comment (evilebottnawi) [reply .babelrc]: We don't need this plugin for async/await:
1. It is inside `@babel/preset-env`
2. We use next branch, in node@8 it is built-in

Comment (evilebottnawi) [reply .babelrc]: Jest don't use babel as the default, we need `babel-jest` package for this

Comment (hiroppy) [reply .babelrc]: no, if we have `.babelrc`, jest uses `babel-jest`. see the documentation

**Comment by hiroppy** [comment]
I want to wait for #2005 merged

File: .babelrc (hunk 1 above)
Comment (evilebottnawi) [reply .babelrc]: :+1:

**Pull request merged**